app/main.py

```diff
@@ -1,4 +1,8 @@
 def get_human_age(cat_age: int, dog_age: int) -> list:
+    if not is_valid_age(cat_age) or not is_valid_age(dog_age):
+        raise ValueError
+    if not isinstance(cat_age, int) or not isinstance(dog_age, int):
+        raise TypeError
```

Comment:
The type check should be performed before any other operations on the input to ensure the input is of the correct type.